addons/godot-next/nodes/behaviors.gd

```diff
@@ -0,0 +1,173 @@
+# author: xdgamestudios
+# license: MIT
+# description: Provides a behavior system API for the nodes. 
+# todo: Refactor code in general to clean and improve performance
+# usage:
+# - Creating:
+#     behaviors = Behaviors.new() # through code
+#     Note:
+#     - To use in the inspector, just add node of type Behaviors.
+# - Adding Behaviors:
+#     behaviors.add_behavior(MyBehavior) # returns instance of MyBehavior
+#     Note:
+#     - Allows only one behavior of each type.
+#     - Returns null if the user assigns an invalid Behavior class. Else returns the added or pre-existing Behavior.
+#     - If behavior already exists current instance is returned.
+# - Checking Behaviors:
+#     behaviors.has_behavior(MyBehavior) # returns true if behavior exists within the collection
+# - Retrieving Behaviors:
+#     behaviors.get_behavior(MyBehavior) # return the instance of the behavior type
+#     Note:
+#     - Returns the Behavior instance or null if it doesn't exist in the collection.
```

Comment:
Again, you can just use this one comment on the same line instead of the other comment and any additional notes.